consensus/polybft/consensus_runtime.go

```diff
@@ -1199,6 +1182,57 @@ func (c *consensusRuntime) BuildRoundChangeMessage(
 	return signedMsg
 }
 
+// getFirstBlockOfEpoch returns the first block of epoch in which provided header resides
+func (c *consensusRuntime) getFirstBlockOfEpoch(epochNumber uint64, latestHeader *types.Header) (uint64, error) {
```

Comment:
it can be any header not necessarily the latest, `latestHeader` => `header`?

Reply:
For now this is only called on `restartEpoch` where we pass the latest header we are synced on.